dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/UsersServiceImpl.java

```diff
@@ -383,71 +384,33 @@ public Map<String, Object> updateUser(User loginUser, int userId,
                                           String phone,
                                           String queue,
                                           int state,
-                                          String timeZone) throws IOException {
+                                          String timeZone) {
         Map<String, Object> result = new HashMap<>();
         result.put(Constants.STATUS, false);
 
         if (resourcePermissionCheckService.functionDisabled()) {
-            putMsg(result, Status.FUNCTION_DISABLED);
-            return result;
+            throw new ServiceException("Function Disabled");
         }
         if (check(result, !canOperator(loginUser, userId), Status.USER_NO_OPERATION_PERM)) {
-            return result;
+            throw new ServiceException("User No Operation Perm");
```

Comment:
we can reuse the `status` here
```suggestion
            throw new ServiceException(Status.USER_NO_OPERATION_PERM.getMsg());
```